src/so_vits_svc_fork/gui.py

```diff
@@ -57,6 +58,54 @@ def delete_preset(name: str) -> dict:
     return load_presets()
 
 
+def get_output_path(input_path: Path) -> Path:
+    # Default output path
+    output_path = input_path.parent / f"{input_path.stem}.out{input_path.suffix}"
+
+    # Increment file number in path if output file already exists
+    file_num = 1
+    while output_path.exists():
+        output_path = (
+            input_path.parent / f"{input_path.stem}.out_{file_num}{input_path.suffix}"
+        )
+        file_num += 1
+    return output_path
+
+
+def get_supported_file_types() -> tuple[tuple[str], ...]:
+    return tuple(
+        [
+            ((extension, f".{extension.lower()}"))
+            for extension in sf.available_formats().keys()
+        ]
+    )
+
+
+def validate_output_file_type(output_path: Path) -> bool:
+    supported_file_types = sorted(
+        [f".{extension.lower()}" for extension in sf.available_formats().keys()]
+    )
+    if not output_path.suffix:
+        sg.popup_ok(
+            "Error: Output path missing file type extension, enter "
+            + "one of the following manually:"
+            + os.linesep
```

Comment:
I think using /n is better if it works, whatever

Reply:
I thought Windows would need `\r\n` but I just tried and it works with just `\n` as well, so I've now switched over.